### PR TITLE
scan: implement mobile for homepage top part

### LINKED ISSUE #2215: Home Page Top Part (Mobile)

### PR DESCRIPTION
**Pull request by tansawit**
Fixed: #2215 

## Implementation details

Implemented mobile version of the top part of Cosmoscan homepage, mainly changing `ChainInfoHighlights.re` and `Searchbar.re`.

## Please ensure the following requirements are met before submitting a pull request:

- [x] The pull request is targeted against the correct target branch
- [x] The pull request is linked to an issue with appropriate discussion and an accepted design OR is linked to a spec that describes the work.
- [x] The pull request includes a description of the implementation/work done in detail.
- [x] The pull request includes any and all appropriate unit/integration tests
- [ ] You have added a relevant changelog entry to `CHANGELOG_UNRELEASED.md`
- [ ] You have re-reviewed the files affected by the pull request (e.g. using the `Files changed` tab in the Github PR explorer)

### CODE REVIEW

File: scan/src/components/ChainInfoHighlights.re [hunk 1]
```diff
@@ -1,15 +1,16 @@
 module Styles = {
   open Css;
 
-  let card =
+  let card = isMobile =>
```
Comment (pzshine) [inline scan/src/components/ChainInfoHighlights.re]: It's no need to use `isMobile` anymore because there is the `Media.mobile` already.

**Review by pzshine — CHANGES_REQUESTED** [review]

File: scan/src/components/SearchBar.re [hunk 2]
```diff
@@ -1,9 +1,9 @@
...
   open Css;
 
-  let container =
+  let container = isMobile =>
     style([
-      maxWidth(`px(600)),
+      maxWidth(isMobile ? `px(344) : `px(600)),
       width(`percent(100.)),
       height(`percent(100.)),
       position(`relative),
```
Comment (evilpeach) [inline scan/src/components/SearchBar.re]: Let's use `Media.mobile()` in style.

File: scan/src/components/SearchBar.re [hunk 3]
```diff
@@ -175,10 +175,11 @@ let reducer = state =>
 
 [@react.component]
 let make = () => {
+  let isMobile = Media.isMobile();
   let ({searchTerm, resultState}, dispatch) =
     React.useReducer(reducer, {searchTerm: "", resultState: Hidden});
 
-  <div className=Styles.container>
+  <div className={Styles.container(isMobile)}>
```
Comment (evilpeach) [inline scan/src/components/SearchBar.re]: No need to pass `isMobile` here.

File: scan/src/components/ChainInfoHighlights.re [hunk 4]
```diff
@@ -82,121 +88,150 @@ let make = (~latestBlockSub: Sub.t(BlockSub.t)) => {
...
-            />;
-          },
-          {
-            let bandPriceInBTC = financial.btcPrice;
-            let usd24HrChange = financial.usd24HrChange;
-            <div className={Styles.withWidth(170)}>
-              <div className=Styles.bandPriceExtra>
+  let isMobile = Media.isMobile();
+
+  let subtitleWidth = isMobile ? 142 : 170;
```
Comment (evilpeach) [inline scan/src/components/ChainInfoHighlights.re]: Let's move this into `withWidth` style.

**Review by evilpeach — CHANGES_REQUESTED** [review]

File: scan/src/components/ChainInfoHighlights.re [hunk 5]
```diff
@@ -32,7 +38,10 @@ module Styles = {
 
   let vFlex = style([display(`flex), flexDirection(`row)]);
 
-  let withWidth = (w: int) => style([width(`px(w))]);
+  let vFlexWithMobile =
```
Comment (evilpeach) [inline scan/src/components/ChainInfoHighlights.re:41]: Let's add `Media.mobile()` on `vFlex` and use it.

File: scan/src/components/ChainInfoHighlights.re [hunk 6]
```diff
@@ -46,6 +55,10 @@ module Styles = {
       zIndex(1),
       opacity(0.4),
     ]);
+
+  let bottom = isMobile => style([marginTop(isMobile ? `px(10) : `zero)]);
```
Comment (evilpeach) [inline scan/src/components/ChainInfoHighlights.re]: Use `Media.mobile`

**Review by evilpeach — CHANGES_REQUESTED** [review]

**Review by pzshine — APPROVED** [review]

**Review by evilpeach — APPROVED** [review]

**Pull request merged**